src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,80 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    private static final int FIRST_PART = 0;
+    private static final int SECOND_PART = 1;
+    private static final String COMMA = ",";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String[] dataFromFile = readFile(fromFileName);
+        int[] supplyAndBuy = infoProcess(dataFromFile);
+        String report = createReport(supplyAndBuy);
+        writeToFile(toFileName, report);
+    }
+
+    private String[] readFile(String fromFileName) {
+        try (BufferedReader reader = new BufferedReader(new FileReader(fromFileName))) {
+            String line;
+            String[] data = new String[0];
+            while ((line = reader.readLine()) != null) {
+                data = append(data, line);
+            }
+            return data;
+        } catch (IOException exception) {
+            throw new RuntimeException("Can`t read data from file " + fromFileName, exception);
+        }
+    }
+
+    private int[] infoProcess(String[] dataFromFile) {
```

Comment:
method name is not related to it's action - consider renaming it to something like "calculateSupplyAndBuy"